src/components/enter-password/tests/enter-password-controller.test.ts

```diff
@@ -47,6 +47,96 @@ describe("enter password controller", () => {
   });
 
   describe("enterPasswordPost", () => {
+    describe("sending MFA code", () => {
+      it("should call the MFA service to send an code when required and able to", async () => {
+        const fakeService: EnterPasswordServiceInterface = {
+          loginUser: sinon.fake.returns({
+            data: {
+              redactedPhoneNumber: "3456",
+              mfaRequired: true,
+              consentRequired: false,
+              latestTermsAndConditionsAccepted: true,
+              mfaMethodVerified: true,
+              mfaMethodType: "SMS",
+              passwordChangeRequired: false,
+            },
+            success: true,
+          }),
+        } as unknown as EnterPasswordServiceInterface;
+
+        const fakeMfaService: MfaServiceInterface = {
+          sendMfaCode: sinon.fake.returns({
+            success: true,
+          }),
+        } as unknown as MfaServiceInterface;
+
+        res.locals.sessionId = "123456-djjad";
+        res.locals.clientSessionId = "00000-djjad";
+        res.locals.persistentSessionId = "dips-123456-abc";
+        req.session.user = {
+          email: "joe.bloggs@test.com",
+        };
+        req.body["password"] = "password";
+
+        await enterPasswordPost(
+          false,
+          fakeService,
+          fakeMfaService
+        )(req as Request, res as Response);
+
+        expect(fakeMfaService.sendMfaCode).to.have.been.called;
```

Comment:
Is it worth adding in the explicit expectations for what this is called with i.e. with an undefined journey type? (Or a separate test that is the negation of the below test, so "should not send a journey type to the MFA service when the user session does not have the reauthenticate property"